astropy/table/table_helpers.py

```diff
@@ -21,22 +21,21 @@ class TimingTables:
     are useful for timing and other API tests.
     """
 
-    def __init__(self, size=1000, masked=False):
+    def __init__(self, rng, size=1000, masked=False):
         self.masked = masked
 
         # Initialize table
         self.table = Table(masked=self.masked)
 
         # Create column with mixed types
-        np.random.seed(12345)
```

Comment:
Looks like with the new fixture, the ability to change seed is gone?

Reply:
Indeed. But if we're going to switch away from the legacy API we're losing the exact reproducibility of test cases anyway.
Quoting numpy docs:
> The RandomState provides access to legacy generators... This class should only be used if it is essential to have randoms that are identical to what would have been produced by previous versions of NumPy.

So, if tests still pass, it means we don't *need* exact reproducibility across this migration, but repeatability still seems important, which is why I *do* set a seed in the fixture.

Reply:
What if someone wants to run tests with different seeds on purpose for some reason?

Reply:
If we want to support that then I think we should *not* have a global fixture for it and we should instead keep using 
`np.random.default_rng` directly as Mridul did in #16086. We could have a parametrized fixture, but then we're back to square one and we don't gain anything over using `rng = np.random.default_rng(0)` directly.

I don't have strong opinions here, I'm honestly fine with how #16086 turned out *without* this fixture, I just didn't want @nstarman's review to go ignored.

Reply:
I don't see this fixture being necessary either. In saving some SLOC we lose flexibility. I'd rather have a bit more lines of code. I wonder what other maintainers think.

Reply:
We can make the fixture reproducible!
See https://github.com/nengo/pytest-rng for an example where the seed can be set.

Reply:
So... based on discussions elsewhere, looks like only removing this `TimingTables` and the tiny refactor `test_item_access.py` are relevant now?